src/Chip/ChipCompat.cs

```diff
@@ -0,0 +1,137 @@
+//
+// ChipCompat.cs
+//
+// Authors:
+//	Rachel Kang  <rachelkang@microsoft.com>
+//com
+// Copyright (C) Microsoft Corporation. All rights reserved.
+//
+
+using System;
+using System.Threading.Tasks;
+using Foundation;
+using CoreFoundation;
+using ObjCRuntime;
+
+#nullable enable
+
+#if !NET
+namespace Chip {
+
+#if !MONOMAC
+	public partial class ChipReadAttributeResult { }
+#endif // !MONOMAC
+
+	[Obsolete ("This class is removed.")]
+	[Register ("CHIPError", SkipRegistration = true)]
+	public class ChipError : NSObject {
+
+		public virtual NSError Create (int errorCode) => throw new InvalidOperationException ();
```

Comment:
Are you suggesting that I replace InvalidOperationException with `PlatformNotSupportedException` or just that I include the constant here?

Assuming you are suggesting both, should I use `PlatformNotSupportedException` here as well? https://github.com/xamarin/xamarin-macios/blob/b016e737dabc2a9705c36d0d941272360c20e650/src/chip.cs#L166

Reply:
my reference: https://github.com/dalexsoto/xamarin-macios/blob/main/src/HomeKit/HMCompat.cs

although now I think I might be totally misinterpreting your comment here...

Reply:
I was thinking maybe I misimplemented this class entirely, and it should be something like the other ones:

```cs
	[Obsolete ("This class is removed.")]
	[Register ("CHIPError", SkipRegistration = true)]
	public class ChipError : NSObject {

		public override IntPtr ClassHandle => throw new InvalidOperationException (Constants.RemovedFromChip);

		protected ChipError (NSObjectFlag t) => throw new InvalidOperationException (Constants.RemovedFromChip);
		protected ChipError (IntPtr handle) => throw new InvalidOperationException (Constants.RemovedFromChip);
		
		public virtual int ConvertToChipErrorCode (NSError errorCode) => throw new InvalidOperationException (Constants.RemovedFromChip);
		public virtual NSError? Create (int errorCode) => throw new InvalidOperationException (Constants.RemovedFromChip);
	
	} /* class ChipError */

```

but then I remembered - while this class was randomly removed, the related enum wasn't. And cecil isn't happy about that

```
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/32bits/iOS/Xamarin.iOS.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/64bits/iOS/Xamarin.iOS.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/32bits/watchOS/Xamarin.WatchOS.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_ios-build/Library/Frameworks/Xamarin.iOS.framework/Versions/git/lib/64bits/tvOS/Xamarin.TVOS.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/Xamarin.Mac/Xamarin.Mac.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
Cecil.Tests.EnumTest.NoAvailabilityOnError("/Users/builder/azdo/_work/1/s/xamarin-macios/_mac-build/Library/Frameworks/Xamarin.Mac.framework/Versions/Current/lib/mono/4.5/Xamarin.Mac.dll"):
Chip.ChipErrorCode.UnsupportedAttribute
Expected: <empty>
But was: < "Chip.ChipErrorCode.UnsupportedAttribute" >
```

I'm not quite sure of the best way to resolve all of the above

Reply:
Those errors are because of this line:

https://github.com/xamarin/xamarin-macios/blob/b016e737dabc2a9705c36d0d941272360c20e650/src/chip.cs#L45

We don't add availability attributes for enum values that represent error codes (it makes any code that uses them unnecessary complex if you want to avoid the warnings).

So the fix is to remove those availability attributes.

Reply:
> Are you suggesting that I replace InvalidOperationException with PlatformNotSupportedException or just that I include the constant here?

I was suggesting that you replace `InvalidOperationException` with `PlatformNotSupportedException`, and then add the constant, so it becomes:

```cs
throw new PlatformNotSupportedException (Constants.RemovedFromChip); 
```

that said, given that HomeKit uses `InvalidOperationException`, I don't think this is a big issue, and no need to change anything.

Reply:
thanks! made a couple of related updates